bot/decorators.py

```diff
@@ -236,3 +238,31 @@ async def wrapped(*args, **kwargs) -> t.Any:
             return await func(*args, **kwargs)
         return wrapped
     return decorator
+
+
+def ensure_duration_in_future(duration_arg: function.Argument) -> t.Callable:
+    """
+    Ensure the duration argument is in the future.
+
+    If the condition fails, a warning is sent to the invoking context.
+
+    `duration_arg` is the keyword name or position index of the parameter of the decorated command
+    whose value is the target duration.
+
+    This decorator must go before (below) the `command` decorator.
+    """
+    def decorator(func: types.FunctionType) -> types.FunctionType:
+        @command_wraps(func)
+        async def wrapper(*args, **kwargs) -> t.Any:
+            bound_args = function.get_bound_args(func, args, kwargs)
+            target = function.get_arg_value(duration_arg, bound_args)
+
+            ctx = function.get_arg_value(1, bound_args)
+
+            if isinstance(target, datetime) and target < arrow.utcnow():
```

Comment:
I think it's because the duration converter returns a `datetime` and `Arrow` does not subclass it.

I'm not sure why the type is being checked though, especially since it's silently ignored. The converter always returns a `datetime`, and this decorator is only intended to be used with that type. Thus, if the type doesn't match then it's a programmer error that should not just be ignored.

One could argue that the type check can still be used to raise a clearer error message but I argue that it would subvert duck typing. That could be especially pertinent in this case since it's conceivable that this decorator will be used with an `Arrow` parameter in the future. As I mentioned earlier, it is not a subclass of `datetime` despite having an identical interface for this purpose.

If the duration can be None then explicitly check for None instead of doing it this way.

Reply:
The failure isn't very silent
![image](https://user-images.githubusercontent.com/69471777/158937118-16592d98-1ad1-4b61-84ac-f3157779d47b.png)

I suppose I could check whether it's not one of None and str instead of whether it's a datetime

Reply:
Oh, right cause you call the original function and it picks up the error.

Reply:
How about `isinstance(duration, (datetime, Arrow))` for future proofing

Reply:
> I suppose I could check whether it's not one of None and str instead of whether it's a datetime

How about 
```py
try:
    if (target < arrow.utcnow()):
        ...
except TypeError:
    func(...)
```

Reply:
Actually this would be a bit better since it's a narrower scope for the caught exception

```py
try:
    is_future = target > arrow.utcnow())
except TypeError:
    func(...)
else:
    if not is_future:
        await ctx.send("!!!!")
```

Reply:
resolved in 61feabb